ocis-pkg/keycloak/client.go

```diff
@@ -90,34 +90,46 @@ func (c *ConcreteClient) SendActionsMail(ctx context.Context, realm, userID stri
 	return c.keycloak.ExecuteActionsEmail(ctx, token.AccessToken, realm, params)
 }
 
-// GetUserByEmail looks up a user by email.
-func (c *ConcreteClient) GetUserByEmail(ctx context.Context, realm, mail string) (*libregraph.User, error) {
+// GetUserByParams looks up a user by the given parameters.
+func (c *ConcreteClient) GetUserByParams(ctx context.Context, realm string, params gocloak.GetUsersParams) (*libregraph.User, error) {
 	token, err := c.getToken(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	users, err := c.keycloak.GetUsers(ctx, token.AccessToken, realm, gocloak.GetUsersParams{
-		Email: &mail,
-	})
+	users, err := c.keycloak.GetUsers(ctx, token.AccessToken, realm, params)
 	if err != nil {
 		return nil, err
 	}
 
 	if len(users) == 0 {
-		return nil, fmt.Errorf("no users found with mail address %s", mail)
+		return nil, fmt.Errorf("no users found")
 	}
 
 	if len(users) > 1 {
-		return nil, fmt.Errorf("%d users found with mail address %s, expected 1", len(users), mail)
+		return nil, fmt.Errorf("%d users found", len(users))
 	}
 
 	return c.keycloakUserToLibregraph(users[0]), nil
 }
 
+// GetUserByEmail looks up a user by email.
+func (c *ConcreteClient) GetUserByEmail(ctx context.Context, realm, mail string) (*libregraph.User, error) {
```

Comment:
Do we still need email getters? Whats the point if they are not unique?

Reply:
I left it there for if someone else could use it, but I can remove it.